core/src/test/java/org/elasticsearch/search/SearchModuleTests.java

```diff
@@ -82,6 +83,98 @@ public void testRegisterQueryParserDuplicate() {
         }
     }
 
+    public void testRegisteredQueries() {
+        //TODO rework test
+        /*SearchModule module = new SearchModule(Settings.EMPTY, new NamedWriteableRegistry());
```

Comment:
I will fix this test once we settle on how things should look, but the idea is to have here a test that verifies which queries get registered, and duplicate all the query names. I think this is the only safe way to make sure that nothing gets lost.